internal/generator/vector/output/splunk/splunk.go

```diff
@@ -1,102 +1,110 @@
 package splunk
 
 import (
-	"fmt"
-	"strings"
+  "fmt"
+  "strings"
 
-	"github.com/openshift/cluster-logging-operator/internal/generator/vector/normalize"
+  "github.com/openshift/cluster-logging-operator/internal/generator/vector/normalize"
 
-	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
+  "github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
 
-	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
-	. "github.com/openshift/cluster-logging-operator/internal/generator"
-	genhelper "github.com/openshift/cluster-logging-operator/internal/generator/helpers"
-	. "github.com/openshift/cluster-logging-operator/internal/generator/vector/elements"
-	vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
-	corev1 "k8s.io/api/core/v1"
+  logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+  . "github.com/openshift/cluster-logging-operator/internal/generator"
+  genhelper "github.com/openshift/cluster-logging-operator/internal/generator/helpers"
+  . "github.com/openshift/cluster-logging-operator/internal/generator/vector/elements"
+  vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+  corev1 "k8s.io/api/core/v1"
 )
 
 var (
-	splunkEncodingJson = fmt.Sprintf("%q", "json")
+  splunkEncodingJson = fmt.Sprintf("%q", "json")
 )
 
 type Splunk struct {
-	ComponentID  string
-	Inputs       string
-	Endpoint     string
-	DefaultToken string
+  ComponentID  string
+  Inputs       string
+  Endpoint     string
+  DefaultToken string
+  Index        string
 }
 
 func (s Splunk) Name() string {
-	return "SplunkVectorTemplate"
+  return "SplunkVectorTemplate"
 }
 
 func (s Splunk) Template() string {
-	return `{{define "` + s.Name() + `" -}}
+  return `{{define "` + s.Name() + `" -}}
 [sinks.{{.ComponentID}}]
 type = "splunk_hec_logs"
 inputs = {{.Inputs}}
 endpoint = "{{.Endpoint}}"
 compression = "none"
 default_token = "{{.DefaultToken}}"
 timestamp_key = "@timestamp"
+{{if .Index -}}
+index = "{{.Index}}"
```

Comment:
This will need to:
* support [templeting](https://vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#index) 
* need to understand what happens when the field is not found
* Update documentation with proper template example

Reply:
Documentation updated https://github.com/openshift/cluster-logging-operator/pull/2177/files#diff-46e6d1852418f35b4db6dde8b446604e9d144d0cb4838d378ed8e2609156bca8R61

Reply:
> This will need to:
> 
> * support [templeting](https://vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#index)
Template can be useful for separating events probably, but i not sure do we really need it.  
> * need to understand what happens when the field is not found
If you mean that will be if we try to send events to not existed index, here that i found:

_Splunk allows for automatic index creation. This means that if you send an event to an index that doesn't exist, Splunk will attempt to create the index for you. 
If automatic creation is disabled, the event might be rejected or placed in a default index, depending on Splunk configuration._

> * Update documentation with proper template example

Documentation updated https://github.com/openshift/cluster-logging-operator/pull/2177/files#diff-46e6d1852418f35b4db6dde8b446604e9d144d0cb4838d378ed8e2609156bca8R61

Reply:
If will be used, not existed field in index template, we will get error in collector: 
`2023-09-26T10:43:55.867961Z ERROR sink{component_kind="sink" component_id=splunk component_type=splunk_hec_logs component_name=splunk}: vector::internal_events::template: Failed to render template for "index". error=Missing fields on event: [".kubernetes.namespacename"] error_type="template_failed" stage="processing" internal_log_rate_limit=true
`